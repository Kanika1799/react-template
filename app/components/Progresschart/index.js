/**
 *
 * Progresschart
 *
 */

import React, { memo } from 'react';
// import PropTypes from 'prop-types'
import styled from 'styled-components';
import { FormattedMessage as T } from 'react-intl';
import 'antd/dist/antd.css';
import { Card, Progress, Row, Col } from 'antd';
const PieCard = styled(Card)`
  .DonutPie {
    width: 200px;
    height: 50px;
  }
  width: 540px;
  margin-left: 20px;
  height: 320px;
  padding: 3px 16px 28px 17px;
  border-radius: 8px;
  margin-top: 90px;
  .HeadingDiv {
    width: 150px;
    .expense {
      width: 89px;
      height: 40px;
      margin: 1px 80px 3px 0;
      font-family: Inter;
      font-size: 12px;
      font-weight: 500;
      font-stretch: normal;
      font-style: normal;
      line-height: 1.25;
      letter-spacing: 0.96px;
      text-align: left;
      color: #1a2940;
    }
    .category {
      width: 83px;
      height: 20px;
      margin: 3px 86px 23px 0;
      font-family: Inter;
      font-size: 12px;
      font-weight: 500;
      font-stretch: normal;
      font-style: normal;
      line-height: 1.25;
      letter-spacing: normal;
      text-align: left;
      color: #3164c6;
    }
  }
  .Payroll {
    height: 20px;
    font-family: Inter;
    font-size: 16px;
    font-weight: 500;
    font-stretch: normal;
    font-style: normal;
    line-height: 1.5;
    letter-spacing: normal;
    text-align: left;
    color: #222222;
  }
  .PieCol {
    margin-top: 90px;
  }
`;
const ColorSpan = styled.div`
  width: 20px;
  height: 0;
  margin: 10px 10px 0 0;
  border: solid 1px #0a36cc;
  background-color: #0a36cc;
`;
const PieDiv = styled.div`
  margin-top: 30px;
  margin-left: 30px;
`;
function Progresschart() {
  return (
    <div data-testid="progresschart">
      <PieDiv>
        <PieCard>
          <Row>
            <Col>
              <div className="HeadingDiv">
                <span className="expense">
                  <T id="my_expenses" />
                </span>
                <span className="category">
                  <T id="category" />
                </span>
              </div>
            </Col>
            <Col className="PieCol">
              <Progress
                className="DonutPie"
                type="circle"
                strokeColor={{
                  '0%': '#0a36cc',
                  '75%': '#46beee'
                }}
                percent={'1496.86'}
              />
            </Col>
            <Col className="PieCol">
              <Row>
                <ColorSpan></ColorSpan>
                <span className="Payroll"></span>
                <T id="payroll" />
              </Row>
              <Row>
                <ColorSpan></ColorSpan>
                <span className="Payroll"></span>
                <T id="leisure" />
              </Row>
              <Row>
                <ColorSpan></ColorSpan>
                <span className="Payroll"></span>
                <T id="monthly" />
              </Row>
              <Row>
                <ColorSpan></ColorSpan>
                <span className="Payroll"></span>
                <T id="uncategorized" />
              </Row>
            </Col>
          </Row>
        </PieCard>
      </PieDiv>
    </div>
  );
}

Progresschart.propTypes = {};

export default memo(Progresschart);
