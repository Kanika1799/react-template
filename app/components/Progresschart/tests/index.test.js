/**
 *
 * Tests for Progresschart
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import Progresschart from '../index';

describe('<Progresschart />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Progresschart />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 Progresschart component', () => {
    const { getAllByTestId } = renderWithIntl(<Progresschart />);
    expect(getAllByTestId('progresschart').length).toBe(1);
  });
});
