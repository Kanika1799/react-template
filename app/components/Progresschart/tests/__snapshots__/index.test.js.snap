// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progresschart /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      data-testid="progresschart"
    >
      <div
        class="Progresschart__PieDiv-fg7fh6-2 eugaig"
      >
        <div
          class="ant-card ant-card-bordered Progresschart__PieCard-fg7fh6-0 dtqMmC"
        >
          <div
            class="ant-card-body"
          >
            <div
              class="ant-row"
            >
              <div
                class="ant-col"
              >
                <div
                  class="HeadingDiv"
                >
                  <span
                    class="expense"
                  >
                    MY EXPENSES
                  </span>
                  <span
                    class="category"
                  >
                    (Category)
                  </span>
                </div>
              </div>
              <div
                class="ant-col PieCol"
              >
                <div
                  class="ant-progress ant-progress-circle ant-progress-status-success ant-progress-show-info ant-progress-default DonutPie"
                >
                  <div
                    class="ant-progress-inner ant-progress-circle-gradient"
                    style="width: 120px; height: 120px; font-size: 24px;"
                  >
                    <svg
                      class="ant-progress-circle"
                      viewBox="0 0 100 100"
                    >
                      <defs>
                        <lineargradient
                          id="ant-progress-gradient-1"
                          x1="100%"
                          x2="0%"
                          y1="0%"
                          y2="0%"
                        >
                          <stop
                            offset="0%"
                            stop-color="#0a36cc"
                          />
                          <stop
                            offset="75%"
                            stop-color="#46beee"
                          />
                        </lineargradient>
                      </defs>
                      <path
                        class="ant-progress-circle-trail"
                        d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                        fill-opacity="0"
                        stroke-linecap="round"
                        stroke-width="6"
                        style="stroke-dasharray: 295.3097094374406px 295.3097094374406px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s;"
                      />
                      <path
                        class="ant-progress-circle-path"
                        d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
                        fill-opacity="0"
                        opacity="1"
                        stroke="url(#ant-progress-gradient-1)"
                        stroke-linecap="round"
                        stroke-width="6"
                        style="stroke: [object Object]; stroke-dasharray: 295.3097094374406px 295.3097094374406px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s, opacity .3s ease 0s;"
                      />
                    </svg>
                    <span
                      class="ant-progress-text"
                    >
                      <span
                        aria-label="check"
                        class="anticon anticon-check"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="check"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 00-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="ant-col PieCol"
              >
                <div
                  class="ant-row"
                >
                  <div
                    class="Progresschart__ColorSpan-fg7fh6-1 gSyAoy"
                  />
                  <span
                    class="Payroll"
                  />
                  Payroll
                </div>
                <div
                  class="ant-row"
                >
                  <div
                    class="Progresschart__ColorSpan-fg7fh6-1 gSyAoy"
                  />
                  <span
                    class="Payroll"
                  />
                  Leisure
                </div>
                <div
                  class="ant-row"
                >
                  <div
                    class="Progresschart__ColorSpan-fg7fh6-1 gSyAoy"
                  />
                  <span
                    class="Payroll"
                  />
                  Monthly
                </div>
                <div
                  class="ant-row"
                >
                  <div
                    class="Progresschart__ColorSpan-fg7fh6-1 gSyAoy"
                  />
                  <span
                    class="Payroll"
                  />
                  Uncategorized
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
